src/services/staticPropsFetcher.ts

```diff
@@ -0,0 +1,15 @@
+import { dehydrate, QueryClient } from 'react-query';
+
+export const staticPropsFetcher = async (
+  key: string | (string | number)[],
+  fetchFunc: () => void,
+) => {
+  const queryClient = new QueryClient();
+  await queryClient.prefetchQuery(key, fetchFunc);
+
+  return {
+    props: {
+      dehydratedState: dehydrate(queryClient),
+    },
+  };
+};
```

Comment:
Ik zou de type voor `fetchFunc` generic maken. Nu zegt ie dat elke function de `() => void` format moet hebben, maar dat is niet helemaal waar denk ik, tenzij react-query dit expliciet vraagt. Dat weet ik niet.

Regel 15/17
https://www.typescriptlang.org/play?#code/PTAEFkHsGMGsEsB2BzAUNANgQwM49AIoCuApgE4CeAwhvCYgC6gDeqAkAA5kkBmJD0ABbFyFUAF5QuComigAFFgBcUxBQA0oAEYqsagJQSAfKABukeABMWAX1R3UPIrIbxIiUJZKCKlslgYSRV0DW1RHZ2hXd1A+AUEAVRxyeSsVRCIAWy1yfRUAeS0AKxIolnZuBiIyD1Y2NkQsTJIVAHIAKUhBDwARSBJW9XYbAG57cJBQAElMjgwSZsYAt0RUEgAPDkgyJmh3HCYD5egABTJIDhwAMX4hcgkpHBk5AB4rjw3AxEt8eQA6AFYMjIHAqZywRCQADuiAA2gBdQziEzgyEwozyVCgUCwEgUFQHMhIZCgAA+CkJxLJoAy2VyCKG2LiQiukRU7yGSJMrGxe0QB1AAEdSJQaHRGA9ECQoYQRdRaPQGPJ9GNsVgoVh4ExhaIxYq-lxeLdhHL5LiNLFjazZCrwtjKtValjsaAuBdQeUXS6vD4-AESJYAMoMf0qH2+fyBeQ60UKxj6RkumyJ0b2MaoSYAFRIB1QR1cp3Olxu8RSzthrSIyTIg1AlerAH0rK14YnmYlq5z0-n4IX3SW7mRMdiK1XyLX6+Qm5YW4nlcZLfEkikE6hbXmQwWzv3jWWR5Oa5oD9PZ8758jF0Jl0OAIyr21AA